receiver/kubeletstatsreceiver/scraper_test.go

```diff
@@ -110,23 +124,22 @@ func TestScraperWithMetadata(t *testing.T) {
 
 			md, err := r.Scrape(context.Background())
 			require.NoError(t, err)
-			require.Equal(t, tt.dataLen, md.DataPointCount())
-
-			for i := 0; i < md.ResourceMetrics().Len(); i++ {
-				rm := md.ResourceMetrics().At(i)
-				for j := 0; j < rm.ScopeMetrics().Len(); j++ {
-					ilm := rm.ScopeMetrics().At(j)
-					require.Equal(t, "otelcol/kubeletstatsreceiver", ilm.Scope().Name())
-					for k := 0; k < ilm.Metrics().Len(); k++ {
-						m := ilm.Metrics().At(k)
-						if strings.HasPrefix(m.Name(), tt.metricPrefix) {
-							_, ok := rm.Resource().Attributes().Get(tt.requiredLabel)
-							require.True(t, ok)
-							continue
-						}
-					}
-				}
-			}
+
+			filename := "test_scraper_with_metadata_" + strings.ReplaceAll(tt.name, " ", "_") + "_expected.yaml"
```

Comment:
Can you please rename all the test cases to have `_` instead of renaming them here? Having spaces in the name is pretty annoying as they are relaced with underscores in the test run output, so it's harder to find which tests failed. Also, we probably shouldn't have `'` in filename like `test_scraper_with_pvc_labels_don't_collect_detailed_labels_expected.yaml`

Reply:
Good idea, I updated the patch to fix this.